src/frontend/app/store/effects/github.effects.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import { Actions, Effect } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { AppState } from '../app-state';
+import { GITHUB_REPO_ENTITY_KEY } from '../types/github.types';
+import { StartRequestAction, WrapperRequestActionSuccess, WrapperRequestActionFailed } from '../types/request.types';
+import { NormalizedResponse } from '../types/api.types';
+import { FetchGitHubRepoInfo, FETCH_GITHUB_REPO } from '../actions/github.actions';
+
+@Injectable()
+export class GithubEffects {
+
+  constructor(
+    private http: Http,
+    private actions$: Actions,
+    private store: Store<AppState>
+  ) {
+  }
+  @Effect() fetchCommit$ =  this.actions$.ofType<FetchGitHubRepoInfo>(FETCH_GITHUB_REPO)
```

Comment:
Might be wrong, but shouldn't all this be handled by using a generic CFStart action?